### PR TITLE
gosamcontrib: add shared and static variants

### PR DESCRIPTION
**Pull request by iarspider**
Not 100% sure if the conflicts are required (they are more of a foolproof rather than real conflicts)

### CODE REVIEW

**Comment by haampie** [comment]
Can you do this with

```
    variant('libs', default='shared,static', values=('shared', 'static'),
            multi=True, description='Build shared libs, static libs or both')
```

and 

```
args += self.enable_or_disable('libs')
```

and conflicts dropped?

**Comment by iarspider** [comment]
@haampie Wow, I didn't know `enable_or_disable` handles multi-value variants. TIL...

File: var/spack/repos/builtin/packages/gosam-contrib/package.py [hunk 1]
```diff
@@ -17,6 +17,24 @@ class GosamContrib(AutotoolsPackage):
     version('2.0', sha256='c05beceea74324eb51c1049773095e2cb0c09c8c909093ee913d8b0da659048d')
     version('1.0', sha256='a29d4232d9190710246abc2ed97fdcd8790ce83580f56a360f3456b0377c40ec')
 
+    variant('libs', default='shared,static', values=('shared', 'static'),
+            multi=True, description='Build shared libs, static libs or both')
+    variant('pic', default=False, description='Build position-independent code')
+
+    conflicts('~shared', when='~static', msg='Please enable at least one of shared or static')
+    conflicts('~static', when='~shared', msg='Please enable at least one of shared or static')
```
Comment (haampie) [inline var/spack/repos/builtin/packages/gosam-contrib/package.py]:
```suggestion
```

**Review by haampie — CHANGES_REQUESTED** [review]

File: var/spack/repos/builtin/packages/gosam-contrib/package.py [hunk 2]
```diff
@@ -17,6 +17,24 @@ class GosamContrib(AutotoolsPackage):
...
     def configure_args(self):
-        args = ["FFLAGS=-std=legacy"]
+        args = []
+        args += self.enable_or_disable('libs')
+
+        if '+pic' in spec:
+            args.extend([
+                'CFLAGS={0}'.format(self.compiler.cc_pic_flag),
+                'CXXFLAGS={0}'.format(self.compiler.cxx_pic_flag),
+                'FFLAGS={0} -std=legacy'.format(self.compiler.f77_pic_flag)
```
Comment (haampie) [inline var/spack/repos/builtin/packages/gosam-contrib/package.py]: Could you do this with `flag_handler`? It's more composable with compiler / command line provided `spack install pkg cflags=... cxxflags=... fflags=...`, grep the code base for examples :upside_down_face:

Comment (iarspider) [reply var/spack/repos/builtin/packages/gosam-contrib/package.py]: Ok.

**Review by haampie — APPROVED** [review]
@spackbot run pipeline

**Pull request merged**